src/abstractdataframe/selection.jl

```diff
@@ -182,6 +182,14 @@ end
 # add a method to funname defined in other/utils.jl
 funname(row::ByRow) = funname(row.fun)
 
+function make_pair_concrete(@nospecialize(x))
+    if x isa Pair
+        return make_pair_concrete(x.first) => make_pair_concrete(x.second)
+    else
+        return x
+    end
+end
```

Comment:
Isn't this equivalent and more idiomatic?
```suggestion
make_pair_concrete(@nospecialize(x::Pair)) =
    make_pair_concrete(x.first) => make_pair_concrete(x.second)
make_pair_concrete(@nospecialize(x)) = x
```

Reply:
I was not sure if it works this way, but it seems it does :).